cl/cl/src/merkle.rs

```diff
@@ -0,0 +1,239 @@
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+pub fn padded_leaves<const N: usize>(elements: &[Vec<u8>]) -> [[u8; 32]; N] {
+    let mut leaves = [[0u8; 32]; N];
+
+    for (i, element) in elements.iter().enumerate() {
+        assert!(i < N);
+        leaves[i] = leaf(element);
+    }
+
+    leaves
+}
+
+pub fn leaf(data: &[u8]) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+    hasher.update(b"NOMOS_MERKLE_LEAF");
```

Comment:
We should consider moving to tag codes if necessary. In Merkle tree constraints, that could save a lot of cycles.

Reply:
Could you elaborate?